From 3e1517c397ebb9d0514008314e03a8df10276b8a Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Fri, 27 Jul 2018 14:29:29 +0200
Subject: [PATCH] lineagehw: Don't send updates to livedisplay driver
 unnecessarily

* The livedisplay driver is smart enough to send all the commands
  that were chosen on panel on by itself, so don't resend it

Change-Id: I51119eb532877f299b7aaad9f2a1d87e0a2344a6
---
 src/org/lineageos/hardware/AdaptiveBacklight.java   | 3 +++
 src/org/lineageos/hardware/AutoContrast.java        | 3 +++
 src/org/lineageos/hardware/ColorEnhancement.java    | 3 +++
 src/org/lineageos/hardware/SunlightEnhancement.java | 4 +++-
 4 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/src/org/lineageos/hardware/AdaptiveBacklight.java b/src/org/lineageos/hardware/AdaptiveBacklight.java
index f9cc2bd..c0c6e72 100644
--- a/src/org/lineageos/hardware/AdaptiveBacklight.java
+++ b/src/org/lineageos/hardware/AdaptiveBacklight.java
@@ -77,6 +77,9 @@ public class AdaptiveBacklight {
         if (sHasNativeSupport) {
             return LiveDisplayVendorImpl.getInstance().setAdaptiveBacklightEnabled(status);
         }
+        if (isEnabled() == status) {
+            return true;
+        }
         return FileUtils.writeLine(FILE_CABC, status ? "1" : "0");
     }
 }
diff --git a/src/org/lineageos/hardware/AutoContrast.java b/src/org/lineageos/hardware/AutoContrast.java
index e048a04..48f42fd 100644
--- a/src/org/lineageos/hardware/AutoContrast.java
+++ b/src/org/lineageos/hardware/AutoContrast.java
@@ -61,6 +61,9 @@ public class AutoContrast {
      * failed; true in any other case.
      */
     public static boolean setEnabled(boolean status) {
+        if (isEnabled() == status) {
+            return true;
+        }
         return FileUtils.writeLine(FILE_ACO, status ? "1" : "0");
     }
 
diff --git a/src/org/lineageos/hardware/ColorEnhancement.java b/src/org/lineageos/hardware/ColorEnhancement.java
index 144b208..3f00b8e 100644
--- a/src/org/lineageos/hardware/ColorEnhancement.java
+++ b/src/org/lineageos/hardware/ColorEnhancement.java
@@ -61,6 +61,9 @@ public class ColorEnhancement {
      * failed; true in any other case.
      */
     public static boolean setEnabled(boolean status) {
+        if (isEnabled() == status) {
+            return true;
+        }
         return FileUtils.writeLine(FILE_CE, status ? "1" : "0");
     }
 }
diff --git a/src/org/lineageos/hardware/SunlightEnhancement.java b/src/org/lineageos/hardware/SunlightEnhancement.java
index 4410f57..d688dfc 100644
--- a/src/org/lineageos/hardware/SunlightEnhancement.java
+++ b/src/org/lineageos/hardware/SunlightEnhancement.java
@@ -96,7 +96,9 @@ public class SunlightEnhancement {
         if (sHasNativeSupport) {
             return LiveDisplayVendorImpl.getInstance().setOutdoorModeEnabled(status);
         }
-
+        if (isEnabled() == status) {
+            return true;
+        }
         return FileUtils.writeLine(FACEMELT_PATH, status ? FACEMELT_MODE : "0");
     }
 
-- 
2.17.1

